app/controllers/candidate_interface/candidate_interface_controller.rb

```diff
@@ -7,9 +7,15 @@ class CandidateInterfaceController < ActionController::Base
     before_action :add_identity_to_log
     layout 'application'
     alias :audit_user :current_candidate
+    before_action :redirect_to_dashboard
 
   private
 
+    def redirect_to_dashboard
+      @application_form = current_application
+      render template: 'candidate_interface/application_form/complete' if current_application.submitted?
```

Comment:
The dashboard will link to other pages, like a new review component, and the pages for accepting/declining/calling up support to make a decision on a course choice. With the line above, a user will be redirected back to the main dashboard page when they go to one of those.

This may also prevent users from reaching candidate-specific static pages e.g. terms of use or stuff accessible from the footer.

I think a callback like this could be the right approach. It feels like we need some kind of namespacing, or whitelisting/blacklisting to determine which pages should redirect to the dashboard post-submission, and which pages should continue working the same (such as static pages). Those could work but I think they may be not very elegant / involve lots of brittle repetition.

Another consideration: we *will* actually open up editing eventually, but it needs to be allowed only after the application has been put into a specific state in order to comply with business rules.

Channeling @tijmenb if he has ideas on how to implement whitelisting of routes or some kind of post-submission namespace so that users can't go back to edit pages.

Reply:
Also, probably just a nit as will have been considered, but this should be a `redirect_to` instead of a `render`.

Reply:
Probably the safest way is to add `before_action :redirect_to_dashboard_if_submitted` to all controllers with a edit interface. We'll just need to make sure we don't forget to add it to everything.

Reply:
Thanks @tijmenb @tvararu

Reply:
Sorted. More cleanup and test required but making progress